app/src/main/java/com/zulip/android/ZulipApp.java

```diff
@@ -287,6 +290,18 @@ public EventsBranch deserialize(JsonElement json, Type typeOfT, JsonDeserializat
                         }
                         Class<? extends EventsBranch> t = EventsBranch.BranchType.fromRawType(invalid);
```

Comment:
since this refactor is to be made in the existing code, i have added a different commit for it